pyfenix/ws_api.py

```diff
@@ -0,0 +1,71 @@
+"""Implementation of the client API layer for websockets"""
+
+import json
+import logging
+import queue
+from typing import Optional, Tuple
+
+import websockets.client
+
+from pyfenix.api import API, NoConnectionError
+from pyfenix.event import Event
+
+class WebsocketsAPI(API):
+    """Fenix client API using websockets"""
+
+    def __init__(self, event_queue: queue.Queue):
+        """
+        :param event_queue: Event queue shared by the event loop and GUI
+        """
+        self._conn: Optional[websockets.client.WebSocketClientProtocol] = None
+        self._queue = event_queue
+        self._server_uri: Optional[str] = None
+
+    async def connect(self, server: Tuple[str, int]) -> None:
+        """
+        Establish a connection to a Fenix server
+
+        An Event.CONN_FAIL event will be generated if the connection fails.
+
+        :param server: (address, port) pair, for example ("127.0.0.1", 21337)
```

Comment:
Maybe say what exceptions can be thrown in doctoring as well?

Reply:
Which exceptions are you thinking of? I didn't encounter any exceptions during testing that weren't handled internally.

Reply:
Ah nevermind, I read line 37 wrong.